packages/dev/core/src/Misc/screenshotTools.ts

```diff
@@ -253,36 +255,56 @@ export function CreateScreenshotUsingRenderTarget(
-                    DumpTools.DumpData(width, height, data, successCallback as (data: string | ArrayBuffer) => void, mimeType, fileName, true, undefined, quality);
-                    texture.dispose();
-                });
-            } else {
-                ApplyPostProcess("pass", texture.getInternalTexture()!, scene, undefined, undefined, undefined, finalWidth, finalHeight).then((texture) => {
-                    engine._readTexturePixels(texture, finalWidth, finalHeight, -1, 0, null, true, false, 0, 0).then((data) => {
-                        DumpTools.DumpData(finalWidth, finalHeight, data, successCallback as (data: string | ArrayBuffer) => void, mimeType, fileName, true, undefined, quality);
+    customizeTexture?.(texture);
+
+    const renderWhenReady = () => {
+        if (texture.isReadyForRendering() && camera.isReady(true)) {
+            engine.onEndFrameObservable.addOnce(() => {
+                if (finalWidth === width && finalHeight === height) {
+                    texture.readPixels(undefined, undefined, undefined, false)!.then((data) => {
+                        DumpTools.DumpData(width, height, data, successCallback as (data: string | ArrayBuffer) => void, mimeType, fileName, true, undefined, quality);
                         texture.dispose();
                     });
-                });
-            }
-        });
+                } else {
+                    ApplyPostProcess("pass", texture.getInternalTexture()!, scene, undefined, undefined, undefined, finalWidth, finalHeight).then((texture) => {
+                        engine._readTexturePixels(texture, finalWidth, finalHeight, -1, 0, null, true, false, 0, 0).then((data) => {
+                            DumpTools.DumpData(
+                                finalWidth,
+                                finalHeight,
+                                data,
+                                successCallback as (data: string | ArrayBuffer) => void,
+                                mimeType,
+                                fileName,
+                                true,
+                                undefined,
+                                quality
+                            );
+                            texture.dispose();
+                        });
+                    });
+                }
+            });
 
+            texture.render(true);
+
+            // re-render the scene after the camera has been reset to the original camera to avoid a flicker that could occur
+            // if the camera used for the RTT rendering stays in effect for the next frame (and if that camera was different from the original camera)
+            scene.incrementRenderId();
+            scene.resetCachedMaterial();
+            engine.setSize(originalSize.width, originalSize.height);
+            camera.getProjectionMatrix(true); // Force cache refresh;
+            scene.render();
+        } else {
+            setTimeout(renderWhenReady, 16);

```

Comment:
Are there observables we could use instead ?

Reply:
I don't think so, because some people are using the screenshot capability without a render loop...